ipyparallel/controller/hub.py

```diff
@@ -985,6 +995,11 @@ def unregister_engine(self, ident, msg):
         content=dict(id=eid, uuid=uuid)
         self.dead_engines.add(uuid)
         
+        #stop the heartbeats
+        self.hearts.pop(uuid, None)
+        self.heartmonitor.responses.discard(uuid)
```

Comment:
I'm also not sure this is the best way to cleanup heartbeats.  If it is, I think it might be best to factor this out into a method...